src/pybind/mgr/dashboard/controllers/smb.py

```diff
@@ -88,6 +89,26 @@
 }
 
 
+def raise_on_failure(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        result = func(*args, **kwargs)
+
+        if result.get('success') is False:
+            msg = 'Operation failed'
+
+            # Extracts the result msg from either of two possible response structures:
+            if 'results' in result:
+                msg = result['results'][0]['msg'] if 'msg' in result['results'][0] else msg
```

Comment:
```suggestion
                msg = result['results'][0].get('msg', msg)
```
Will check for `'msg'` in `result['results'][0]` and if exists, get that value otherwise defaults to `msg` value set above